src/pages/calculate.tsx

```diff
@@ -75,18 +97,28 @@ export const getServerSideProps = async (
       },
     };
   } catch (e) {
-    // If we encounter any invalid/bad data and throw an error, redirect instead of trying to fall back gracefully. Otherwise,
-    // users may be confused as to why certain params did not take effect.
+    // TypeScript doesn't support type annotations on catch
+    const error = e as Error;
+    const statusCode = HTTP_STATUS_CODES.HTTP_STATUS_BAD_REQUEST;
+    const description = error.message ?? 'One or more query parameters are invalid. Please check the URL you entered.';
     return {
-      redirect: {
-        destination: '/',
-        permanent: false,
+      props: {
+        fonts,
+        initialState: initialFormState,
+        error: {
+          code: statusCode,
+          reasonPhrase: REASON_PHRASES[statusCode] as string,
+          description,
+        },
       },
     };
   }
 };
 
 const Calculate: NextPage<CalculatePageProps> = (props) => {
+  if (props.error) {
+    return <ErrorPage {...props.error} />;
+  }
```

Comment:
As a future enhancement, it would be nice to lazily load this component. Shouldn't be too big of a deal since the `/calculate` route is server-rendered anyway, so it won't impact SEO or ranking.